### DISCUSSION TITLE
Implementing RTL CSS Styling when RTL direction detected

### DISCUSSION THREAD

**Discussion by halwesit** [directus/directus#18565] [Feature Requests]
### Summary

Hey Everyone,
I have just rewritten some CSS code of the Directus admin dashboard, that makes Directus support RTL.

### Basic Example

I think we have 2 main options to make Directus Support RTL internally:

1. Detecting the Directus local and based on the locale direction we can import the customized RTL styling.
2. Adding a toggle RTL Button to the Project Settings by activating it the dashboard became RTL.

### Motivation

Although there are about 12 languages with RTL writing scripts, such as 
- Kurdish (Sorani)
- Arabic.
- Aramaic.
- Hebrew.
- Persian/Farsi.
- Urdu.
- ... and others

### Detailed Design

For now, copy the code below and go to the Project Settings from your Directus dashboard, and Paste the code inside the Custom CSS field, then save it.

RTL CSS Styling:

```
@import url('https://fonts.googleapis.com/css2?family=Vazirmatn:wght@500&display=swap');

* {
 direction: rtl; 
  --family-sans-serif: 'Vazirmatn', Sans-serif;
}

.project-info {
  padding-right: 20px;
}
.public-view .title-box .title {
  margin-top: 2px;
  margin-right: 16px;
}
.public-view .notice {
  display: flex;
  color: var(--foreground-subdued);
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
}
.public-view .v-icon {
  --v-icon-color: var(--foreground-subdued);
  margin-right: 4px;
}
.v-icon.left {
  margin-left: 8px;
}
.v-list .v-list-item-icon:not(:only-child):first-child,
.v-list-item .v-list-item-icon:not(:only-child):first-child {
  margin-left: 8px;
}
.v-list-item-icon:not(:only-child):first-child {
  margin-right: 0;
}
.input {
  display: flex;
  flex-direction: row-reverse !important;
}
.v-divider span.wrapper .v-icon {
  margin-left: 4px;
  transform: translateY(-1px);
}
a.v-list-item.link {
  padding: 0px !important;
  margin: var(--v-list-item-margin-nav) !important;
}
.headline {
  right: 0 !important;
}
.v-button.secondary.rounded {
  margin-left: 8px !important;
  margin-right: 0 !important;
}
.title-container.full {
  padding: 0 !important;
  margin: 0 !important;
  margin-left: 20px !important;
  padding-left: 20px !important;
  margin-right: 8px !important;
}
.v-icon i {
  display: -webkit-box !important;
}
.sidebar-detail .title {
  right: 52px !important;
}
.sidebar-detail .close {
  left: 0 !important;
  right: auto !important;
}
.v-divider.inlineTitle span.wrapper {
  margin-left: 8px !important;
  margin-right: 0 !important;
}
.v-card-actions {
  justify-content: space-evenly !important;
}
.drag-container {
  margin-right: 20px !important;
  margin-left: 0 !important;
}
span.v-icon.collection-icon {
  margin-left: 8px;
  margin-right: 0 !important;
}
a.v-list-item.link.version {
  display: none !important;
}
.permissions-overview-row .name .actions {
  margin: 0 8px 0 0 !important;
}
.permissions-overview-row
  .permissions-overview-toggle
  + .permissions-overview-toggle {
  margin-right: 20px !important;
  margin-left: 0 !important;
}
.permissions-overview-header .v-icon + .v-icon {
  margin-left: 0 !important;
  margin-right: 20px !important;
}
span.instant-save {
  margin-left: 0 !important;
  margin-right: 4px !important;
}
.table-row .cell.select {
  justify-content: flex-end !important;
}
.search-input {
  flex-direction: row-reverse !important;
}
.v-badge.search-badge {
  margin: 0 0px 0px 8px !important;
}
.search-input .icon-filter {
  margin: 0 8px 0 4px !important;
}
.v-drawer .cancel {
  top: 32px;
  right: -70px !important;
  left: unset !important;
  margin: 0 10px 0 0px !important;
}
.end .sort-selector {
  margin: 0 8px 0 8px !important;
}
dt {
  margin: 0 0px 0 8px !important;
}
.user-box .avatar {
  margin: 0 0 0 20px !important;
}
.start .label {
  margin: 0 4px 0 0 !important;
}
span.v-icon.edit {
  margin: 0 0 0 4px !important;
}
.many-to-one .v-input .append {
  flex-direction: row-reverse !important;
}
.display-translations .icon {
  margin: 0 8px;
}
.v-list-group .items {
  padding: 0 24px 0 12px !important;
}
span.description-dot {
  margin: 0 0px 0 8px !important;
}
span.v-icon.has-click.regenerate-icon {
  margin: 0 0 0 4px !important;
}
.v-input .input .append {
  margin: 0 0px 0 8px !important;
} 
```

### Must Have:
- Importing the custom RTL CSS code

### Could Have:
- An option for using RTL Styling for none RTL locales as well as in opposite.

### Drawbacks

Theres NO drawbacks

### Alternatives

//

### Adoption Strategy

As I explained in the basic example, we can add this feature just by writing a couple of lines of code, because already Directus has a feature of detecting locale directions in the translation interface [translation interface RTL](https://github.com/directus/directus/pull/14665), so we can import the custom CSS styling for RTL when the locale direction is RTL.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
